libs/auth/src/openssl.ts

```diff
@@ -208,15 +208,23 @@ export async function createCert({
   signingPrivateKey: FilePathOrBuffer;
   signingPrivateKeyPassword: string;
 }): Promise<Buffer> {
+  // TODO: Instead of using a throwaway private key to generate the cert signing request and
+  // injecting the public key we want using -force_pubkey, have the relevant HSM (Java Card, TPM,
+  // etc.) generate the cert signing request
+  const throwawayPrivateKey = await openssl([
+    'ecparam',
+    '-genkey',
+    '-name',
+    'prime256v1',
+    '-noout',
+  ]);
   const certSigningRequest = await openssl([
     'req',
     '-new',
     '-config',
     opensslConfig,
     '-key',
-    signingPrivateKey,
-    '-passin',
-    `pass:${signingPrivateKeyPassword}`,
+    throwawayPrivateKey,
```

Comment:
While the output of this function is still the same (given the `-force_pubkey` below), the previous code was quite misleading. We'd ideally be using the private key of the public key to sign to generate the CSR, but until we figure out how to do that with Java Cards, we're at least now being more explicit about the fact that we're using a throwaway private key to generate the CSR.